interfaces.d.ts

```diff
@@ -115,11 +115,16 @@ declare module './dist/vaadin-router' {
     redirect: string;
   }
 
+  interface RouteWithAnimation extends BaseRoute {
```

Comment:
Rather then adding `RouteWithAnimate` alternative to the union, I think we should add `animate` to existing interfaces.

Consider a corner case route object that only specifies required keys. The idea here is that TypeScript will require `path` and either of `action`, `bundle`, `children`, `component`, and `redirect`. Each one from this list completes a valid route, and a route not having any of them would not compile.

So we have `Route` defined as a union of `RouteWithChildren`, `RouteWithComponent` and such specifically to implement this alternative requirement behaviour for those route keys.

However, I think `animate` should not be a required alternative, since a route with only `path` and `animate` does not make much sense: it specifies no particular content to render and animate.

Thinking a bit more, it seems rather impractical to combine `animate` with redirects and such. I suppose, we should only have certain route types animatable, for example, only routes with `children` or `component`.

Also, note that the animate key supports custom classnames object.

```typescript
  interface AnimateCustomClasses {
    enter?: string;
    leave?: string;
  }

  interface AnimatableRoute extends BaseRoute {
    animate?: boolean | AnimateCustomClasses;
  }

  // ...

  interface RouteWithChildren extends AnimatableRoute {
    children: Route[] | ChildrenFn;
  }

  interface RouteWithComponent extends AnimatableRoute {
    component: string;
  }
```

Reply:
Please check my next commit.